internal/commands/app/diff_test.go

```diff
@@ -139,26 +135,102 @@ Removed Dependencies
   - debug@4.3.1
 Modified Dependencies
   * underscore@1.9.1 -> underscore@1.9.2
-`, out.String())
+`
+		t.Run("with include node modules set it should diff function dependencies", func(t *testing.T) {
+			out, ui := mock.NewUI()
+			cmd := &CommandDiff{diffInputs{LocalPath: "testdata/dependencies", IncludeNodeModules: true}}
+			assert.Equal(t, nil, cmd.Handler(nil, ui, cli.Clients{Realm: realmClient}))
+
+			assert.Equal(t, diffStr, out.String())
+		})
+
+		t.Run("with include dependencies set it should diff function dependencies", func(t *testing.T) {
+			out, ui := mock.NewUI()
+			cmd := &CommandDiff{diffInputs{LocalPath: "testdata/dependencies", IncludeDependencies: true}}
+			assert.Equal(t, nil, cmd.Handler(nil, ui, cli.Clients{Realm: realmClient}))
+
+			assert.Equal(t, diffStr, out.String())
+		})
+
+		t.Run("with include package json set it should diff function dependencies", func(t *testing.T) {
+			out, ui := mock.NewUI()
+			cmd := &CommandDiff{diffInputs{LocalPath: "testdata/dependencies", IncludePackageJSON: true}}
+			assert.Equal(t, nil, cmd.Handler(nil, ui, cli.Clients{Realm: realmClient}))
+
+			assert.Equal(t, diffStr, out.String())
+		})
 	})
 
-	t.Run("should return error with include dependencies set and diff dependencies returns an error", func(t *testing.T) {
+	t.Run("should return error when more than one dependencies flag is set", func(t *testing.T) {
 		_, ui := mock.NewUI()
 
-		realmClient := mock.RealmClient{}
+		t.Run("when include node modules and include package json are both set", func(t *testing.T) {
+			cmd := &CommandDiff{diffInputs{LocalPath: "testdata/dependencies", IncludeNodeModules: true, IncludePackageJSON: true}}
+			assert.Equal(t, errors.New("cannot use both \"include-node-modules\" and \"include-package-json\" at the same time"), cmd.inputs.Resolve(nil, ui))
```

Comment:
[optional/style] to avoid needing to escape `"`, you can use:
```suggestion
			assert.Equal(t, errors.New(`cannot use both "include-node-modules" and "include-package-json" at the same time`), cmd.inputs.Resolve(nil, ui))
```

if you go this route, just make sure to update the other instances.  it generally helps the readability